Automattic/wp-calypso
# Plugins: Remove plugins main flux store

#### Changes proposed in this Pull Request

After #48665 reduxified the last bits that were using the plugins flux store directly, it's now time to remove the store. This PR takes care of that by removing all the unnecessary flux action dispatches and then removing the actual store with its tests and README. 

In addition, this PR fills the gaps by reduxifying a couple of features that were still partially relying on the legacy flux store indirectly:

* Plugin upload
* Premium plugin installation

Part of #24180.

#### Testing instructions

* Get several Jetpack sites (including multisite with subsites) with some plugins for testing. 
* Test the following scenarios with and without an empty Redux store.
* Thoroughly test lists of plugins in all plugin scenarios; compare against production and verify that they still work the same way when updating, installing, or removing plugins, as well as toggling auto-updates and activating/deactivating plugins:
  * `/plugins`
  * `/plugins/manage`
  * `/plugins/manage` - for multiple selected plugins (bulk actions are enabled by clicking the "Edit All" button)
  * `/plugins/manage/:site` - for a single plugin
  * `/plugins/manage/:site` - for multiple selected plugins (bulk actions are enabled by clicking the "Edit All" button)
  * `/plugins/:plugin/:site`
  * `/plugins/:plugin` - make sure to try multisite and its subsites, too.
* Specifically try uploading an old version of a plugin and verify everything behaves the same way before, during, and after the upload.
* Specifically test updating an old plugin and verify everything behaves the same way before, during, and after the update.
* Specifically try `/plugins/setup/:site` for a new site on a free plan, on a new site that you just purchased a paid plan with, and on an old site with a paid plan and verify everything works the same way as in production.
* Further test anything in plugins you consider not listed above.
* Verify all tests pass.
